engine/src/flutter/impeller/renderer/backend/vulkan/descriptor_pool_vk_unittests.cc

```diff
@@ -71,50 +49,27 @@ TEST(DescriptorPoolRecyclerVKTest, ReclaimDropsDescriptorPoolIfSizeIsExceeded) {
     std::vector<std::unique_ptr<DescriptorPoolVK>> pools;
     for (auto i = 0u; i < 33; i++) {
       auto pool = std::make_unique<DescriptorPoolVK>(context);
-      pool->AllocateDescriptorSets({}, *context);
+      pool->AllocateDescriptorSets({}, 0, *context);
       pools.push_back(std::move(pool));
     }
   }
 
-  // See note above.
-  for (auto i = 0u; i < 2; i++) {
-    auto waiter = fml::AutoResetWaitableEvent();
-    auto rattle = fml::ScopedCleanupClosure([&waiter]() { waiter.Signal(); });
-    {
-      UniqueResourceVKT<fml::ScopedCleanupClosure> resource(
-          context->GetResourceManager(), std::move(rattle));
-    }
-    waiter.Wait();
-  }
-
   auto const called = GetMockVulkanFunctions(context->GetDevice());
   EXPECT_EQ(
       std::count(called->begin(), called->end(), "vkCreateDescriptorPool"),
       33u);
-  EXPECT_EQ(std::count(called->begin(), called->end(), "vkResetDescriptorPool"),
-            33u);
 
   // Now create 33 more descriptor pools and observe that only one more is
   // allocated.
   {
-    std::vector<std::unique_ptr<DescriptorPoolVK>> pools;
+    std::vector<std::shared_ptr<DescriptorPoolVK>> pools;
```

Comment:
Why this change?

Reply:
before we used to vend unique objects that had internal shared state. now we just vend the shared state.